app/src/main/kotlin/com/softteco/template/utils/LockScreenOrientation.kt

```diff
@@ -0,0 +1,19 @@
+package com.softteco.template.utils
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.platform.LocalContext
+
+@Composable
+fun LockScreenOrientation(orientation: Int) {
+    val activity = LocalContext.current.getActivity() ?: return
+    val originalOrientation = activity.resources.configuration.orientation
+
+    DisposableEffect(orientation) {
+        activity.requestedOrientation = orientation
+
+        onDispose {
+            activity.requestedOrientation = originalOrientation
+        }
+    }
+}
```

Comment:
It looks like most of the screens do not support landscape orientation.
It seems to me that in this case it would be better to do so:

1. Add to Manifest
```kotlin
android:screenOrientation="portrait"
```
2. Rename and edit the method:

```kotlin
@Composable
fun UnlockScreenOrientation() {
    val activity = LocalContext.current.getActivity() ?: return
    val originalOrientation = activity.resources.configuration.orientation

    DisposableEffect(Unit) {
        activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR

        onDispose {
            activity.requestedOrientation = originalOrientation
        }
    }
}
```
3. To call UnlockScreenOrientation() method only on screens that support landscape orientation

Reply:
Apparently, my example will include support for landscape orientation on all screens... But I think the idea is clear)) I'm not sure if this can be done.

Reply:
Yes, there's more boilerplate code now, and I've thought about that. But:
1. The ratio of affected to unaffected screens is such because the project contains only basic functions. But it is being designed to be scalable. And the ratio will change when the project is used for its intended purpose.
2. To provide good UX, it is better to have support for both orientations. So right now we can consider the lack of landscape orientation support as a problem of a specific screen. And this implementation is just a temporary hack. When we go back to refactoring these screens and implement landscape orientation support, the changes will not affect the already correct screens.